demo/combo-box-basic-demos.html

```diff
@@ -6,91 +6,155 @@
+            comboBox.value = 'Option one';
           });
         </script>
       </template>
     </vaadin-demo-snippet>
 
+    <h3>Value change event</h3>
+    <vaadin-demo-snippet id="allow-custom-value" when-defined="vaadin-combo-box">
+      <template preserve-content>
+        <vaadin-combo-box label="Label"></vaadin-combo-box>
+        <div id="output">Selected value:</div>
+        <script>
+          window.addDemoReadyListener('#allow-custom-value', function(document) {
+            const comboBox = document.querySelector('vaadin-combo-box');
+            const output = document.querySelector('#output');
+            comboBox.items = ['Option one', 'Option two'];
+            comboBox.addEventListener('change', function(event) {
+              output.textContent = event.target.value;
+            });
+          });
+        </script>
+      </template>
+    </vaadin-demo-snippet>
 
-    <h3>Allow Custom Values</h3>
+    <h3>Allow custom values</h3>
     <p>Allow the user to set any value for the field in addition to selecting a value from the dropdown menu.</p>
     <vaadin-demo-snippet id="combo-box-basic-demos-allow-custom-values" when-defined="vaadin-combo-box">
       <template preserve-content>
-        <vaadin-combo-box label="Element" allow-custom-value></vaadin-combo-box>
-        <p>Selected value: <span></span></p>
-
+        <vaadin-combo-box label="Fruit" allow-custom-value></vaadin-combo-box>
+        <p>Selected value: <span></span></span></p>
         <script>
           window.addDemoReadyListener('#combo-box-basic-demos-allow-custom-values', function(document) {
             const comboBox = document.querySelector('vaadin-combo-box');
             const span = document.querySelector('span');
+            const fruits = ['Apple', 'Orange', 'Banana'];
+            comboBox.items = fruits;
+            comboBox.value = span.textContent = 'Apple';
+
+            comboBox.addEventListener('change', function() {
+              span.textContent = comboBox.value;
+            });
+          });
+        </script>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Storing custom values</h3>
+    <vaadin-demo-snippet id="combo-box-second-custom-value" when-defined="vaadin-combo-box">
+      <template preserve-content>
+        <vaadin-combo-box label="Project" allow-custom-value></vaadin-combo-box>
+        <p>Selected id: <span></span></span></p>
+        <script>
+          window.addDemoReadyListener('#combo-box-second-custom-value', function(document) {
+            const comboBox = document.querySelector('vaadin-combo-box');
+            const span = document.querySelector('span');
 
+            comboBox.items = fetchProjects();
+            comboBox.itemValuePath = 'id';
+            comboBox.itemLabelPath = 'name';
+
+            if (comboBox.items.length > 0) {
+              comboBox.value = comboBox.items[0].name;
+              span.textContent = comboBox.items[0].id;
```

Comment:
The value change listener should handle this (when defined before the value change).

Reply:
I wanted to have an initial value, but if you think it's not important to have it I can delete it.

Reply:
I mean that if you place a "value-changed" listener to the comboBox _before_ setting its `value`, the event listener will fire and set the `span.textContent` automatically, so you don't need to repeat the code

Reply:
Done